src/main/telemetry/ibus_shared.c

```diff
@@ -57,120 +79,460 @@ typedef enum {
 
+static uint8_t getSensorID(ibusAddress_t address) {
+    //all checks are done in theAddressIsWithinOurRange
+    uint32_t index = address - ibusBaseAddress;
+    return telemetryConfig()->flysky_sensors[index];
+}
 
+static uint8_t getSensorLength(uint8_t sensorID) {
+    if (sensorID == IBUS_SENSOR_TYPE_PRES || (sensorID >= IBUS_SENSOR_TYPE_GPS_LAT && sensorID <= IBUS_SENSOR_TYPE_ALT_MAX)) {
+        return IBUS_4BYTE_SESNSOR;
+    }
+#if defined(TELEMETRY_IBUS_EXTENDED)
+    if (sensorID == IBUS_SENSOR_TYPE_GPS_FULL) {
+        return 14;
+    }
+    if (sensorID == IBUS_SENSOR_TYPE_VOLT_FULL) {
+        return 10;
+    }
+    if (sensorID == IBUS_SENSOR_TYPE_VOLT_FULL) {
+        return 12;
+    }
+#endif
+    return IBUS_2BYTE_SESNSOR;
+}
 
-static uint8_t transmitIbusPacket(uint8_t *ibusPacket, size_t payloadLength)
+static uint8_t transmitIbusPacket()
 {
-    uint16_t checksum = calculateChecksum(ibusPacket, payloadLength + IBUS_CHECKSUM_SIZE);
+    uint8_t frameLength = sendBuffer[0];
+    if (frameLength == INVALID_IBUS_ADDRESS) {
+        return 0;
+    }
+    uint8_t payloadLength = frameLength - IBUS_CHECKSUM_SIZE;
+    uint16_t checksum = calculateChecksum(sendBuffer);
     for (size_t i = 0; i < payloadLength; i++) {
-        serialWrite(ibusSerialPort, ibusPacket[i]);
+        serialWrite(ibusSerialPort, sendBuffer[i]);
     }
     serialWrite(ibusSerialPort, checksum & 0xFF);
     serialWrite(ibusSerialPort, checksum >> 8);
-    return payloadLength + IBUS_CHECKSUM_SIZE;
+    return frameLength;
 }
 
-static uint8_t sendIbusDiscoverSensorReply(ibusAddress_t address)
+static void setIbusDiscoverSensorReply(ibusAddress_t address)
 {
-    uint8_t sendBuffer[] = { 0x04, IBUS_COMMAND_DISCOVER_SENSOR | address};
-    return transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+    sendBuffer[0] = IBUS_HEADER_FOOTER_SIZE;
+    sendBuffer[1] = IBUS_COMMAND_DISCOVER_SENSOR | address;
 }
 
-static uint8_t sendIbusSensorType(ibusAddress_t address)
+static void setIbusSensorType(ibusAddress_t address)
 {
-    uint8_t sendBuffer[] = {0x06,
-                            IBUS_COMMAND_SENSOR_TYPE | address,
-                            sensorAddressTypeLookup[address - ibusBaseAddress],
-                            0x02
-                           };
-    return transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+    uint8_t sensorID = getSensorID(address);
+    uint8_t sensorLength = getSensorLength(sensorID);
+    sendBuffer[0] = IBUS_HEADER_FOOTER_SIZE + 2;
+    sendBuffer[1] = IBUS_COMMAND_SENSOR_TYPE | address;
+    sendBuffer[2] = sensorID;
+    sendBuffer[3] = sensorLength;
 }
 
-static uint8_t sendIbusMeasurement(ibusAddress_t address, uint16_t measurement)
-{
-    uint8_t sendBuffer[] = { 0x06, IBUS_COMMAND_MEASUREMENT | address, measurement & 0xFF, measurement >> 8};
-    return transmitIbusPacket(sendBuffer, sizeof(sendBuffer));
+static uint16_t getVoltage(){
```

Comment:
Coding style is K&R style: Opening brace on new line for function definitions, at the end of line (with whitespace) for everything else.

Reply:
Done - I have use different editors with different formatting configuration.